backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchSearchPredicate.java

```diff
@@ -31,9 +33,10 @@ static ElasticsearchSearchPredicate from(ElasticsearchSearchIndexScope<?> scope,
 			throw log.cannotMixElasticsearchSearchQueryWithOtherPredicates( predicate );
 		}
 		ElasticsearchSearchPredicate casted = (ElasticsearchSearchPredicate) predicate;
-		if ( !scope.hibernateSearchIndexNames().equals( casted.indexNames() ) ) {
-			throw log.predicateDefinedOnDifferentIndexes( predicate, casted.indexNames(),
-					scope.hibernateSearchIndexNames() );
+		Set<String> difference = notInTheOtherSet( scope.hibernateSearchIndexNames(), casted.indexNames() );
```

Comment:
This is called very often: every time we add a predicate/sort/projection/agregation to a query or to a composite predicate/sort. Perhaps that would be premature optimization, but I'd suggest looking into some optimization...

E.g. if you make sure the index names are sorted (which is easy to achieve in `AbstractSearchIndexScope`, even if we keep a `LinkedHashSet`) , you can compute the difference in `O(n)` without any `Set` allocation in the best and most common case (empty difference).